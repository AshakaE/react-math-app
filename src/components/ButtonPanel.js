import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = ({ clickHandler }) => (
  <>
    <div className="flex w-[700px] h-[100px] justify-around">
      <Button name="AC" clickHandler={clickHandler} />
      <Button name="+/-" clickHandler={clickHandler} />
      <Button name="%" clickHandler={clickHandler} />
      <Button color="orange" name="รท" clickHandler={clickHandler} />
    </div>
    <div className="flex w-[700px] h-[100px] justify-around">
      <Button name="7" clickHandler={clickHandler} />
      <Button name="8" clickHandler={clickHandler} />
      <Button name="9" clickHandler={clickHandler} />
      <Button color="orange" name="*" clickHandler={clickHandler} />
    </div>
    <div className="flex w-[700px] h-[100px] justify-around">
      <Button name="4" clickHandler={clickHandler} />
      <Button name="5" clickHandler={clickHandler} />
      <Button name="6" clickHandler={clickHandler} />
      <Button color="orange" name="-" clickHandler={clickHandler} />
    </div>
    <div className="flex w-[700px] h-[100px] justify-around">
      <Button name="1" clickHandler={clickHandler} />
      <Button name="2" clickHandler={clickHandler} />
      <Button name="3" clickHandler={clickHandler} />
      <Button color="orange" name="+" clickHandler={clickHandler} />
    </div>
    <div className="flex w-[700px] h-[100px] justify-between">
      <Button wide="wide" name="0" clickHandler={clickHandler} />
      <Button name="." clickHandler={clickHandler} />
      <Button color="orange" name="=" clickHandler={clickHandler} />
    </div>
  </>
);

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
