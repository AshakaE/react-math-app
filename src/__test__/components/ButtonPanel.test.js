import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../../components/ButtonPanel';

const test = () => {};

describe('ButtonPanel', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ButtonPanel clickHandler={test} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
