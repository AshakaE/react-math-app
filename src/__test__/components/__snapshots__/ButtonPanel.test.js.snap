// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel renders correctly 1`] = `
Array [
  <div
    className="flex w-[700px] h-[100px] justify-around"
  >
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none orange "
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>,
  <div
    className="flex w-[700px] h-[100px] justify-around"
  >
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none orange "
      onClick={[Function]}
      type="button"
    >
      *
    </button>
  </div>,
  <div
    className="flex w-[700px] h-[100px] justify-around"
  >
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none orange "
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>,
  <div
    className="flex w-[700px] h-[100px] justify-around"
  >
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none orange "
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>,
  <div
    className="flex w-[700px] h-[100px] justify-between"
  >
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray wide"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="w-1/4 border border-gray-400 text-2xl focus:outline-none orange "
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>,
]
`;
